TheMovieDb/Modules/Home/HomeRemoteDataManager.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  HomeRemoteDataManager.swift
+//  TheMovieDb
+//
+//  Created by Javier Cueto on 14/11/21.
+//  
+//
+
+import Foundation
+
+final class HomeRemoteDataManager: HomeRemoteDataManagerInputProtocol {
+    
+    var remoteRequestHandler: HomeRemoteDataManagerOutputProtocol?
+    private let group = DispatchGroup()
+    private var movies: [MovieGroupSections: [Movie]] = [:]
+    
+    func fetchMovies() {
+        fetchData(typeMovieSection: .popular)
+        fetchData(typeMovieSection: .upcoming)
+        fetchData(typeMovieSection: .topRated)
+        fetchData(typeMovieSection: .playingNow)
+        fetchData(typeMovieSection: .trending)
+        group.notify(queue: .main) {
+            self.remoteRequestHandler?.fetchedMovies(self.movies)
+        }
+    }
+    
+    private func fetchData(typeMovieSection: MovieGroupSections) {
+        group.enter()
+        MovieAPI.shared.fetchData(endPoint: typeMovieSection.path, completion: {(response: Result<Movies, Error>) in
```

Comment:
MovieAPI should not be called directly instead add it as a dependency from the class or the method, this will allow you to test this latter

```swift
let service: MovieAPI = .shared  // Ideally should be a protocol this & var to test this from init

    private func fetchData(typeMovieSection: MovieGroupSections) {
        group.enter()
        service.fetchData(endPoint: typeMovieSection.path, completion: {(response: Result<Movies, Error>) in
```

or can be as a parameter into the method also

Reply:
Now  using this  //     var service: APIMoviesProtocol? // and injecting to the object in the wireFrame class